tools/RAiDER/models/credentials.py

```diff
@@ -0,0 +1,144 @@
+                                 \n"url"   : "{host}",\
+                                 \n"key"   : "{key}",\
+                                 \n"email" : "{uid}"\
+                                 \n}}
+                                """,
+                        'help_url' : 'https://confluence.ecmwf.int/display/WEBAPI/Access+ECMWF+Public+Datasets#AccessECMWFPublicDatasets-key'
+                        },
+        'netrc' :       {'api' : """\
+                                \nmachine {host}\
+                                \n        login {uid}\
+                                \n        password {key}\
+                                """,
+                       'help_url': 'https://wiki.earthdata.nasa.gov/display/EL/How+To+Access+Data+With+cURL+And+Wget'
+                       }
+                }
+
+# system environmental variables for API credentials
+'''
+ENV variables in cdsapi and ecmwapir
+
+cdsapi ['cdsapirc'] : CDSAPI_KEY [UID:KEY], 'CDSAPI_URL'
+ecmwfapir [ecmwfapirc] : 'ECMWF_API_KEY', 'ECMWF_API_EMAIL','ECMWF_API_URL'  
+
+'''
+
+# Check if API enviroments exists
+def _check_envs(model):
+    if model in ('ERA5', 'ERA5T'):
+        uid = os.getenv('RAIDER_ECMWF_ERA5_UID')
+        key = os.getenv('RAIDER_ECMWF_ERA5_API_KEY')
+        host = API_URLS['cdsapirc']
+
+    elif model in ('HRES'):
+        uid = os.getenv('RAIDER_HRES_EMAIL') 
+        key = os.getenv('RAIDER_HRES_API_KEY')
+        host = os.getenv('RAIDER_HRES_URL')
+        if host is None:
+            host = API_URLS['ecmwfapirc']
+
+    elif model in ('GMAO'):
+        uid = os.getenv('EARTHDATA_USERNAME') # same as in DockerizedTopsApp
+        key = os.getenv('EARTHDATA_PASSWORD')
+        host = API_URLS['netrc']
+
+    else: # for HRRR
+        uid, key, host = None, None, None
+
+    return uid, key, host
+
+# Check and write MODEL API_RC_FILE for downloading weather model data
+def check_api(model: str,
+              UID: str = None,
+              KEY: str = None,
+              output_dir : str = '~/',
+              update_flag: bool = False) -> None:
+
+    # Weather model API filename
+    # typically stored in home dir as hidden file
+    api_filename = API_FILENAME[model]
+
+    # Get API credential from os.env if UID/KEY are not inserted
+    if UID is None and KEY is None:
+        UID, KEY, URL = _check_envs(model)
+    else:
+        URL = API_URLS[api_filename]
+
+    # Get hidden ext for Windows
+    hidden_ext = '_' if system()=="Windows" else '.'
+
+    # skip below if model is HRRR as it does not need API
+    if api_filename:    
+        # Check if the credential api file exists
+        api_filename_path = Path(output_dir) / (hidden_ext + api_filename)
+        api_filename_path = api_filename_path.expanduser()
+
+        # if update flag is on, overwrite existing file 
+        if update_flag:
+            api_filename_path.unlink() if update_flag==True and api_filename_path.exists() else None
```

Comment:
I believe this can be simplified to:
```suggestion
        if update_flag is True:
            api_filename_path.unlink(missing_ok=True)
```
https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink